nibabel/volumeutils.py

```diff
@@ -1334,26 +1334,37 @@ def _ftype4scaled_finite(tst_arr, slope, inter, direction='read',
     tst_arr = np.atleast_1d(tst_arr)
     slope = np.atleast_1d(slope)
     inter = np.atleast_1d(inter)
-    warnings.filterwarnings('ignore', '.*overflow.*', RuntimeWarning)
-    try:
-        for ftype in OK_FLOATS[def_ind:]:
-            tst_trans = tst_arr.copy()
-            slope = slope.astype(ftype)
-            inter = inter.astype(ftype)
+    overflow_filter = ('error', '.*overflow.*', RuntimeWarning)
+    for ftype in OK_FLOATS[def_ind:]:
+        tst_trans = tst_arr.copy()
+        slope = slope.astype(ftype)
+        inter = inter.astype(ftype)
+        try:
             if direction == 'read':  # as in reading of image from disk
                 if slope != 1.0:
-                    tst_trans = tst_trans * slope
+                    # Keep warning contexts small to reduce the odds of a race
```

Comment:
Is the repetition really necessary?  The test array is always two elements, so the whole thing adds up to (2 multiplies 2 adds) or  (2 subtracts and 2 divides), with an if statement.  This is pretty damn fast:

```
[ins] In [1]: import numpy as np                                                                                       
[ins] In [2]: ii = np.iinfo(np.int64)                                                                                  
[ins] In [3]: tst_arr = np.array([ii.min, ii.max], dtype=np.int64)    
[nav] In [6]: def do_mult(tst_arr): 
         ...:     if True: 
         ...:         res = tst_arr - 1 
         ...:         res = tst_arr / 0.1 
         ...:                                                                                                          
[ins] In [7]: timeit do_mult(tst_arr)                                                                                  
1.79 µs ± 25 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Presumably this is a slight overestimate, with the function call overhead.  Compare to:

```
[ins] In [8]: timeit tst_arr / 0.1                                                                                     
988 ns ± 22.4 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)                              
```

So - I think the time window reduction is very small.

Reply:
Okay, I see that I failed to follow all the logic and realize this is only ever used on a 2-element array. I'll wrap the entire branch.